### ISSUE TITLE
Responsive navbar is 2px shorter when content is hidden.

### ISSUE THREAD

**Issue by puzrin** [twbs/bootstrap#21604]
Example: http://v4-alpha.getbootstrap.com/components/navbar/#nav, see first snippet.

At narrow screen navbar block has height 8 + 38 + 8 px (54px). At wide screen height become 8 + 40 + 8 px (56px). Reproducible in latest FF/Chromium, Ubuntu 14.04.

I think height should not change. Any quick-tips how to fix?

**Comment by bardiharborow** [comment]
Both `.navbar-nav` and `.navbar-toggler` are `40px` tall, but `.navbar` does not seem to count `.navbar-toggler`'s `2px` of border when calculating its height. This is why I hate CSS. @mdo

**Comment by rishabh-rajgarhia** [comment]
>**TL;DR** <br> Increasing line-height of `.navbar-brand` from `1.5 (inherited from body)` to `1.6` makes the height of `.navbar-brand` `40px (4 + 32 + 4)` and in turn `.navbar` to `56px (8 + 40 + 8)`.


This is probably because height of `.navbar-brand` sums up to `38px (4 + 30 + 4)`  and `.nav-link`  sums up to `40px (8 + 24 + 8)` making the height of the `.navbar` `56px (8 + 40 + 8)` in wider screens as height of `.nav-link` is more than that of `.navbar-brand`. 

Whereas in collapsed state, `.navbar` takes the height of `.navbar-brand`, i.e., `38px` making the height of `.navbar` `54px (8 + 38 + 8)`. `.navbar` does not count the height of `.navbar-toggler (40px)` as it is positioned as `absolute` so it does not affect the flow of other elements.

Possible solution can be increasing line-height of `.navbar-brand` from `1.5 (inherited from body)` to `1.6` making the height of `.navbar-brand` `40px (4 + 32 + 4)` and in turn `.navbar` to `56px (8 + 40 + 8)`.

_Here is an [example](https://www.codeply.com/go/tBkifTbMHh)._

You guys are the expert. If @mdo and @bardiharborow are okay with this solution, I can quickly create a pull request for this. Love the work the whole community does and I would love to contribute in any way I can :)

**Comment by vanduynslagerp** [comment]
I implemented a solution here #21721 that consist in computing the `navbar-brand` padding to make it of same height as the other component in the `navbar`.

The difference is due to mismatched font-size between `navbar-band` and `nav-link`, and the height being the result of `font-size * line-height`.

`nav-link` is `1 (font-size) * 1.5 (line-height)` with a padding of `0.5rem`, for a total height of `2.5rem`.
`navbar-brand` is `1.25 (font-size) * 1.5 (line-height)`, for a total of `1.875rem` before padding.

Then we can calculate the padding for `navbar-brand`:
```scss
+$navbar-brand-font-size:            $font-size-lg !default;
 +// Compute the navbar-brand padding-y so the navbar-brand will have the same height as navbar-text and nav-link
 +$nav-link-height:                   $navbar-brand-font-size * $line-height-base !default;
 +$navbar-brand-height:               ($font-size-base * $line-height-base + $nav-link-padding-y * 2) !default;
 +$navbar-brand-padding-y:            ($navbar-brand-height - $nav-link-height) / 2 !default;
````
